lib/active_job/traffic_control/throttle.rb

```diff
@@ -30,7 +30,7 @@ def throttling_lock_key(job)
         around_perform do |job, block|
           if self.class.job_throttling.present?
             lock_options = {
-              resources: self.class.job_throttling[:threshold],
+              resources: self.class.lock_resources(job, self.class.job_throttling),
```

Comment:
Per Principal of Least Responsibility, I'd suggest just passing `self.class.job_throttling[:threshold]` as the second parameter, not the entire job configuration. The method only needs the threshold.